NGit/NGit.Util.IO/ThrowingPrintWriter.cs

```diff
@@ -84,7 +85,9 @@ public override void Write(char[] cbuf, int off, int len)
 			@out.Write(cbuf, off, len);
 		}
 
-		/// <exception cref="System.IO.IOException"></exception>
+        public override Encoding Encoding { get; }
```

Comment:
It's not clear why we're doing this. It appears to always be null, right?

Reply:
The interface demanded it existed, so it needed to be there.

Reply:
Yeah, but the interface expects an object to be returned. This always returns null. I don't really know the purpose of this type (GitHub is obstinately refusing to show me more context), but perhaps we should either be delegating to a proxy or raising an exception `(UpsupportedOperation maybe? )`, rather than returning null. If it was test code, I'd be unconcerned, but this is in the main library code. Is returning null definitely the right thing to do here?

Reply:
For example, could you return `@out.Encoding`, assuming `@out` is a proxy?

Reply:
Ah, I see what you mean. Everything else is a proxy for `@out` so it makes sense for this method to be the same. I'll sort it.